mm2src/mm2_bitcoin/spv_validation/src/conf.rs

```diff
@@ -0,0 +1,173 @@
+use crate::helpers_validation::SPVError;
+use crate::work::{DifficultyAlgorithm, RETARGETING_INTERVAL};
+use chain::{BlockHeader, BlockHeaderBits};
+use primitives::hash::H256;
+use serde::Deserialize;
+use std::num::NonZeroU64;
+use std::str::FromStr;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct StartingBlockHeader {
+    // Starting block height
+    pub height: u64,
+    // Hash of the starting block header.
+    pub hash: Option<String>,
+    // Time of the starting block header.
+    pub time: u32,
+    // Bits of the starting block header.
+    pub bits: Option<u32>,
+}
+
+fn validate_btc_max_stored_headers_value(max_stored_block_headers: u64) -> Result<(), SPVError> {
+    if RETARGETING_INTERVAL > max_stored_block_headers as u32 {
+        return Err(SPVError::StartingBlockHeaderError(format!(
+            "max_stored_block_headers {max_stored_block_headers} must be greater than retargeting interval {RETARGETING_INTERVAL}",
+        )));
+    }
+
+    Ok(())
+}
+
+fn validate_btc_starting_header_height(coin: &str, height: u64) -> Result<(), SPVError> {
+    let is_retarget = height % RETARGETING_INTERVAL as u64;
+    if is_retarget != 0 {
+        return Err(SPVError::WrongRetargetHeight {
+            coin: coin.to_string(),
+            expected_height: height - is_retarget,
+        });
+    }
+
+    Ok(())
+}
+
+fn validate_btc_starting_header_bits_hash(coin: &str, header: &Option<StartingBlockHeader>) -> Result<(), SPVError> {
+    if let Some(header) = header {
+        if header.bits.is_none() {
+            return Err(SPVError::BlockHeaderBitsError {
+                coin: coin.to_string(),
+                height: header.height,
+            });
+        }
+
+        if header.hash.clone().unwrap_or_default().is_empty() {
+            return Err(SPVError::BlockHeaderBitsError {
+                coin: coin.to_string(),
+                height: header.height,
+            });
+        }
+    }
+
+    Ok(())
+}
+
+/// SPV headers verification parameters
+#[derive(Clone, Debug, Deserialize)]
+pub struct BlockHeaderValidationParams {
+    pub difficulty_check: bool,
+    pub constant_difficulty: bool,
+    pub difficulty_algorithm: Option<DifficultyAlgorithm>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct SPVConf {
+    /// Where to start block headers sync from.
+    pub starting_block_header: Option<StartingBlockHeader>,
```

Comment:
`starting_block_header` was also non-optional, why make it optional now?

Reply:
same comment as above